### ISSUE TITLE
Inference detects COCO class labels instead of my custom class labels  

### ISSUE THREAD

**Issue by ifahim** [google/automl#484]
I am having good progress in training on efficientdet up to now. I am training custom data with 7 class + 1 class which is negative class (no bbox). I double checked the annotation file and they do have my custom classes.  I prepared data, train and inference as follows: 

```
!python dataset/create_coco_tfrecord.py --logtostderr --image_dir=../../train/  --object_annotations_file=../../train/train.json --output_file_prefix="tfrecord/train"
!python dataset/create_coco_tfrecord.py --logtostderr --image_dir=../../valid/  --object_annotations_file=../../valid/valid.json --output_file_prefix="tfrecord/valid"
!python dataset/create_coco_tfrecord.py --logtostderr --image_dir=../../test/  --object_annotations_file=../../test/test.json --output_file_prefix="tfrecord/test"


!mkdir model_dir/

!python main.py --mode=train_and_eval \
    --training_file_pattern=tfrecord/train-*.tfrecord  \
    --validation_file_pattern=tfrecord/valid-*.tfrecord  \
    --model_name={MODEL} \
    --model_dir=model_dir/{MODEL}-finetune \
    --ckpt={MODEL} \
    --train_batch_size=8 \
    --eval_batch_size=8 --eval_samples={images_per_epoch}  \
    --num_examples_per_epoch={images_per_epoch}  --num_epochs=1  \
    --hparams="num_classes=8,moving_average_decay=0,precision=mixed_float16"

!python model_inspect.py --runmode=saved_model --model_name=efficientdet-d0\
     --ckpt_path=model_dir/{MODEL}-finetune --saved_model_dir=savedmodeldir --hparams="num_classes=8,moving_average_decay=0"

!python model_inspect.py --runmode=saved_model_infer \
  --model_name=efficientdet-d0 \
  --saved_model_dir=savedmodeldir \
  --input_image=ep.jpg --output_image_dir=serve_image_out 
```

**The model detects the perfect bounding boxes but the associated class label in the images are  the COCO class label instead of my custom label.** 
Am I missing any step here?

**Comment by Samjith888** [comment]
Replace this line `coco_id_mapping` with ur custom classes and its ID's.
[coco_id_mapping](https://github.com/google/automl/blob/master/efficientdet/inference.py#L42)

@mingxingtan will you add a flag to pass the custom labels and id's while doing inference ?

**Comment by mingxingtan** [comment]
You can define the label_id_mapping in your voc.yaml  file. I updated README to include an example.

**Comment by ifahim** [comment]
@mingxingtan  and @Samjith888, thanks.

**Comment by github2016-yuan** [comment]
@ifahim 
 I do similar thing with you.  I extract one class (bus) from coco dataset so I get a floder of bus images and a json file of bbox info about bus images. Then I convert json file and bus images to tfrecords. It works well and I get tfrecords for training  efficient-d0.

I use cmd below to train my detector of bus
```
--mode=train    
--training_file_pattern=cocotfrecordbus/train/*    
--model_name=efficientdet-d0    
--model_dir=cocobusmodels      
--backbone_ckpt=efficientnet-b0      
--train_batch_size=8     
--num_examples_per_epoch=3952 
--num_epochs=10      
--hparams="num_classes=1,moving_average_decay=0,precision=mixed_float16"
```
After training 10 epochs, I get ckpt file and convert it to pb file.At last, I use pb file for inference.
But it detects nothing,just empty bbox.

I use tensorboard to visualize my training log, everything seems well. Loss gradually decreases.
You say "The model detects the perfect bounding boxes..." so I wonder if you can give me some advice?

ps: I will train more epochs later to see if it is helpful?

**Comment by github2016-yuan** [comment]
After 90 epochs, the pb file still can not detect nothing when inference. The final loss is 0.58, not very larger. So, strange about that.

**Comment by ifahim** [comment]
@github2016-yuan, please follow the advice:
```
Create a config file for the PASCAL VOC dataset called voc_config.yaml and put this in it.

  num_classes: 20
  var_freeze_expr: '(efficientnet|fpn_cells|resample_p6)'
 moving_average_decay:0

  label_id_mapping: {0: background, 1: aeroplane, 2: bicycle, 3: bird, 4: boat, 5: bottle, 6: bus, 7: car, 8: cat, 9: chair, 10: cow, 11: diningtable, 12: dog, 13: horse, 14: motorbike, 15: person, 16: pottedplant, 17: sheep, 18: sofa, 19: train, 20: tvmonitor}

```

then call: 

```
--mode=train    
--training_file_pattern=cocotfrecordbus/train/*    
--model_name=efficientdet-d0    
--model_dir=cocobusmodels      
--backbone_ckpt=efficientnet-b0      
--train_batch_size=8     
--num_examples_per_epoch=3952 
--num_epochs=10      
--hparams=voc_config.yaml

```
